test_conformance/api/test_queue.cpp

```diff
@@ -59,3 +59,63 @@ int test_queue_flush_on_release(cl_device_id deviceID, cl_context context,
 
     return success ? TEST_PASS : TEST_FAIL;
 }
+
+int test_multi_queue_flush_on_release(cl_device_id deviceID, cl_context context,
+                                      cl_command_queue defaultQueue,
+                                      int num_elements)
+{
+    cl_int err;
+
+    // Create A command queue
+    cl_command_queue queue_A = clCreateCommandQueue(context, deviceID, 0, &err);
+    test_error(err, "Could not create command queue A");
+
+    // Create B command queue
+    cl_command_queue queue_B = clCreateCommandQueue(context, deviceID, 0, &err);
+    test_error(err, "Could not create command queue B");
+
+    // Create a kernel
+    clProgramWrapper program;
+    clKernelWrapper kernel;
+    const char *source = "void kernel test(){}";
+    err = create_single_kernel_helper(context, &program, &kernel, 1, &source,
+                                      "test");
+    test_error(err, "Could not create kernel");
+
+    // Enqueue the kernel on queue_A and obtain event to synchronize with
+    // queue_B
+    size_t gws = num_elements;
+    clEventWrapper event_A;
+    err = clEnqueueNDRangeKernel(queue_A, kernel, 1, nullptr, &gws, nullptr, 0,
+                                 nullptr, &event_A);
+    test_error(err, "Could not enqueue kernel");
+
+    // Enqueue the kernel on queue_B using event_A for synchronization and
+    // create event_B to track completion
+    clEventWrapper event_B;
+    err = clEnqueueNDRangeKernel(queue_B, kernel, 1, nullptr, &gws, nullptr, 1,
+                                 &event_A, &event_B);
+    test_error(err, "Could not enqueue kernel");
+
+    // Release queue_A, which performs an implicit flush to issue any previously
+    // queued OpenCL commands
+    err = clReleaseCommandQueue(queue_A);
+    test_error(err, "clReleaseCommandQueue failed");
+
+    err = clFinish(queue_B);
+    test_error(err, "clFinish failed");
+
+    // Wait for kernel to execute since the queue must flush on release
+    bool success = poll_until(2000, 50, [&event_B]() {
+        cl_int status;
+        cl_int err = clGetEventInfo(event_B, CL_EVENT_COMMAND_EXECUTION_STATUS,
+                                    sizeof(cl_int), &status, nullptr);
```

Comment:
Since we've called `clFinish` on `queue_B` here, the `event_B` must already be `CL_COMPLETE`.